File: backend/supabaseClient.js
// Supabase Client Initialization (Browser & Node.js)
import { createClient } from '@supabase/supabase-js';

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL || 'https://yhgfdzwelegsmxixmuun.supabase.co';
const supabaseKey = process.env.NEXT_PUBLIC_SUPABASE_PUBLISHABLE_KEY || process.env.SUPABASE_ANON_KEY || 'sb_publishable_BcyUOzXvvkaYCqV0GrvbNA_V0RToaF8';

export const supabase = createClient(supabaseUrl, supabaseKey);

File: backend/supabase_client.py
from typing import Optional
from backend.config import settings

def get_supabase_client():
    """
    Initializes and returns the Supabase client using credentials from .env.
    """
    if not settings.SUPABASE_URL or not (settings.SUPABASE_SERVICE_ROLE_KEY or settings.SUPABASE_ANON_KEY):
        raise ValueError(
            "Supabase credentials missing. Please set SUPABASE_URL and SUPABASE_ANON_KEY / SUPABASE_SERVICE_ROLE_KEY in .env"
        )
    
    try:
        from supabase import create_client, Client
        key = settings.SUPABASE_SERVICE_ROLE_KEY or settings.SUPABASE_ANON_KEY
        client: Client = create_client(settings.SUPABASE_URL, key)
        return client
    except ImportError:
        raise ImportError(
            "The 'supabase' Python package is not installed. Run 'pip install supabase' to enable Supabase integration."
        )
